kv/memberlist/memberlist_client_test.go

```diff
@@ -234,14 +234,28 @@ func casWithErr(ctx context.Context, kv *Client, key string, updateFn func(*data
 	return kv.CAS(ctx, key, fn)
 }
 
+var localhostIP *net.IPAddr
+var localhostBindAddrs []string
+var localhostBindAddrsOnce sync.Once
+
+func setLocalhostBindAddrs() {
+	var err error
+	localhostIP, err = net.ResolveIPAddr("ip4", "localhost")
+	if err != nil {
+		localhostBindAddrs = []string{"127.0.0.1"}
+	}
+	localhostBindAddrs = []string{localhostIP.String()}
+}
```

Comment:
Is this a particularly expensive operation? I'm wondering if we can simplify the code a little and just have a `getLocalhostAddr()` method that resolves the address each time.

Reply:
I mean it's a DNS lookup so guessing about a millisecond. Let me see.

Reply:
I think let's simplify this for now and if we notice it being a problem in the future, we can re-introduce this more optimised way of doing things.

Reply:
Just realized I can make it substantially simpler with a helper method, I can also back out the sync.Once, let me know what you think

Reply:
Looks good to me, thanks